src/Data/Unrestricted/Linear.hs

```diff
@@ -48,6 +49,9 @@ data Unrestricted a where
 unUnrestricted :: Unrestricted a ->. a
 unUnrestricted (Unrestricted a) = a
 
+uliftA2 :: (a -> b -> c) -> Unrestricted a ->. Unrestricted b ->. Unrestricted c
+uliftA2 f (Unrestricted a) (Unrestricted b) = Unrestricted (f a b)
```

Comment:
rename: `liftA2`. This module is imported qualified.

Reply:
I'm not sure I think `liftA2` is the right name for this function, it suggests that the argument ought to be `a ->. b ->. c` (since this is what's expected of Data and Control `liftA2`)

Reply:
Well, `liftA2` is alright. Though `lift2` is fine with me too.

 `Unrestricted` is pretty special because it's a functor from the category of unrestricted functions to the category of linear functions. We don't really have an abstraction for that. This is saying that it preserves (monoidal) products, which is sort of like applicative (if we don't decide that _applicative_ is not only monoidal but also endo).

I don't care either way to be honest :slightly_smiling_face: . I just want the `u` to go as I want linear-base to use namespaces for namespacing rather than name conventions.

Reply:
I'm happy with `lift2`

Reply:
Similar to this, it would also be nice to have an unary version `(a -> b) -> Unrestricted a ->. Unrestricted b`.